recipes/opencv/4.x/conanfile.py

```diff
@@ -70,6 +70,9 @@ def configure(self):
         self.options["libtiff"].jpeg = self.options.with_jpeg
         self.options["jasper"].with_libjpeg = self.options.with_jpeg
 
+        if self.settings.os == "Android":
+            self.options.with_openexr = False  # required because this forces linkage to libc++_shared.so
```

Comment:
I assume `parallel` with tbb or openmp is invalid on android? perhaps there should be an exception to let consumers know?

Reply:
As far as I understand OpenCV's CMakeLists.txt, they have several options for parallel execution, filtered internally by several conditions like platform. TBB has a filter for iOS and WinRT (VISIBLE_IF NOT IOS AND NOT WINRT), OpenMP has no filter defined (VISIBLE_IF TRUE), therefore I assume it is legal for Android? pthreads is available for all non-Windows platforms(VISIBLE_IF NOT WIN32 OR MINGW).

Finally, all options are prioritized and the "best" of the enabled and available is selected:

```cmake
ocv_build_features_string(parallel_status EXCLUSIVE
  IF HAVE_TBB THEN "TBB (ver ${TBB_VERSION_MAJOR}.${TBB_VERSION_MINOR} interface ${TBB_INTERFACE_VERSION})"
  IF HAVE_HPX THEN "HPX"
  IF HAVE_OPENMP THEN "OpenMP"
  IF HAVE_GCD THEN "GCD"
  IF WINRT OR HAVE_CONCURRENCY THEN "Concurrency"
  IF HAVE_PTHREADS_PF THEN "pthreads"
  ELSE "none")
status("")
status("  Parallel framework:" "${parallel_status}")
```
 Therefore, I would assume an exception rule would not be easy to define, the cmake file disables unsupported options silently and shows in the summary the used one.

BTW, WITH_PTHREADS_PF is ON by default as most general fallback, I think it would be good to do the same in the recipe?

Reply:
beautiful answer 🌪️ , if opencv does a best effort, I think the correct implementation is perfectly adequate. from a consumer perspective if I told opencv through Conan or CMake directly this will have very similar behaviors which matches my preference!

Reply:
So, this is ok for you? You marked lines 73-74 (openexr exclusion), any question about this?

Reply:
Nope! Feel free to make this resolved